### DISCUSSION TITLE
@theme-ui/color chaining functions, getting hex value

### DISCUSSION THREAD

**Discussion by jayliu50** [system-ui/theme-ui#1855] [Q&A]
Hi! I'm just curious how to do stuff like chaining functions, getting hex values for colors. The methods that I expect to work don't work:

```javascript

// trying to get hex value
desaturate('red', 0.5).toString(); // nope
desaturate('red', 0.5).hex(); // nope
desaturate('red', 0.5).hex; // nope

// trying to chain
lighten(desaturate('red', 0.5), 0.3); // nope

```

Thanks!

**Comment by hasparus** [comment]
Functions from `theme-ui/color` are mostly 2nd order — they return function taking theme object and only then return the resulting color.

i.e.

```
desaturate("primary", 0.5)({
  colors: { primary: "#30f" }
})
```

Take a look at this:
https://github.com/system-ui/theme-ui/blob/a6034456e33a14fb0eb3fa09b97feb32c6f9d37f/packages/color/src/index.ts#L52

**Reply by jayliu50** [reply]
Got it. Thank you so much for your reply. Yeah, I guess chaining the functions was my main intent.

**Reply by hasparus** [reply]
_@theme-ui/color_ is mostly here for convenience in common use cases. If you need to do something more complex you can always grab some functions from _polished_ (_@theme-ui/color_ depends on it so you already have it in node_modules).

```ts
import { darken, desaturate } from 'polished';
import { getColor } from '@theme-ui/color';

<div
  sx={{
    color: (theme) => {
      const primary = getColor(theme, "primary"); // the raw value you specified in your theme
      
      return darken(0.2, desaturate(0.25, primary));
    }
  }}
/>
```

**Reply by jayliu50** [reply]
Awesome! This is great. Thank you again for helping me. And thank you for your work.
